x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/BlobCacheTestUtils.java

```diff
@@ -0,0 +1,337 @@
+        if (fileSize % rangeSize > 0) {
+            numberOfRanges++;
+        }
+        if (numberOfRanges == 0) {
+            numberOfRanges++;
+        }
+        return numberOfRanges;
+    }
+
+    /**
+     * Generates a sorted set of non-empty and non-contiguous random ranges that could fit into a file of a given maximum length.
+     */
+    public static SortedSet<ByteRange> randomRanges(long length) {
+        final SortedSet<ByteRange> randomRanges = new TreeSet<>();
+        for (long i = 0L; i < length;) {
+            long start = randomLongBetween(i, Math.max(0L, length - 1L));
+            long end = randomLongBetween(start + 1L, length); // +1 for non empty ranges
+            randomRanges.add(ByteRange.of(start, end));
+            i = end + 1L + randomLongBetween(0L, Math.max(0L, length - end)); // +1 for non contiguous ranges
+        }
+        return randomRanges;
+    }
+
+    public static SortedSet<ByteRange> mergeContiguousRanges(final SortedSet<ByteRange> ranges) {
+        return ranges.stream().collect(TreeSet::new, (gaps, gap) -> {
+            if (gaps.isEmpty()) {
+                gaps.add(gap);
+            } else {
+                final ByteRange previous = gaps.pollLast();
+                if (previous.end() == gap.start()) {
+                    gaps.add(ByteRange.of(previous.start(), gap.end()));
+                } else {
+                    gaps.add(previous);
+                    gaps.add(gap);
+                }
+            }
+        }, (gaps1, gaps2) -> {
+            if (gaps1.isEmpty() == false && gaps2.isEmpty() == false) {
+                final ByteRange last = gaps1.pollLast();
+                final ByteRange first = gaps2.pollFirst();
+                if (last.end() == first.start()) {
+                    gaps1.add(ByteRange.of(last.start(), first.end()));
+                } else {
+                    gaps1.add(last);
+                    gaps2.add(first);
+                }
+            }
+            gaps1.addAll(gaps2);
+        });
+    }
+
+    public static void assertCacheFileEquals(CacheFile expected, CacheFile actual) {
+        MatcherAssert.assertThat(actual.getLength(), equalTo(expected.getLength()));
+        MatcherAssert.assertThat(actual.getFile(), equalTo(expected.getFile()));
+        MatcherAssert.assertThat(actual.getCacheKey(), equalTo(expected.getCacheKey()));
+        MatcherAssert.assertThat(actual.getCompletedRanges(), equalTo(expected.getCompletedRanges()));
+    }
+
+    public static long sumOfCompletedRangesLengths(CacheFile cacheFile) {
+        return cacheFile.getCompletedRanges().stream().mapToLong(ByteRange::length).sum();
+    }
+
+    /**
+     * A {@link BlobContainer} that can read a single in-memory blob.
+     * Any attempt to read a different blob will throw a {@link FileNotFoundException}
+     */
+    public static BlobContainer singleBlobContainer(final String blobName, final byte[] blobContent) {
```

Comment:
I think we can leave this method (and also `singleSplitBlobContainer`)  in the searchable snapshots plugin